ui/pages/confirmations/components/edit-gas-fee-popover/edit-gas-item/useCustomTimeEstimate.js

```diff
@@ -73,12 +72,19 @@ export const useCustomTimeEstimate = ({
   ) {
     waitTimeEstimate = Number(customEstimatedTime?.upperTimeBound);
   } else if (
-    Number(maxPriorityFeePerGas) >= Number(medium.suggestedMaxPriorityFeePerGas)
+    gasFeeEstimates &&
+    Number(maxPriorityFeePerGas) >=
+      Number(gasFeeEstimates.medium?.suggestedMaxPriorityFeePerGas)
   ) {
-    waitTimeEstimate = high.minWaitTimeEstimate;
-  } else {
-    waitTimeEstimate = low.maxWaitTimeEstimate;
+    waitTimeEstimate = gasFeeEstimates.high?.minWaitTimeEstimate;
```

Comment:
does this and the next condition not need `gasFeeEstimates &&` as well? Alternatively, perhaps `gasFeeEstimates?.` could work on all of them

Reply:
good suggestion ! added optional chaining in the equation